config/ompi_check_libhcoll.m4

```diff
@@ -47,6 +47,19 @@ AC_DEFUN([OMPI_CHECK_HCOLL],[
                               [ompi_check_hcoll_happy="yes"],
                               [ompi_check_hcoll_happy="no"])
 
+           AS_IF([test "$ompi_check_hcoll_happy" = "yes"],
+                 [
+                     CPPFLAGS=$coll_hcoll_CPPFLAGS
+                     LDFLAGS=$coll_hcoll_LDFLAGS
+                     LIBS=$coll_hcoll_LIBS
+                     AC_CHECK_FUNC(hcoll_context_free,
```

Comment:
Using AC_CHECK_FUNCS instead of AC_CHECK_FUNC will add HAVE_function symbol automatically to config.h, no need to pass DHAVE_func